src/cli/format.test.ts

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023-Present The Pepr Authors
+
+import { jest, describe, it, expect, beforeEach } from "@jest/globals";
+import { formatWithPrettier } from "./format.helpers";
+import { promises as fs } from "fs";
+import { resolveConfig, format } from "prettier";
+import { ESLint } from "eslint";
+
+jest.mock("fs", () => ({
+  promises: {
+    readFile: jest.fn(),
+    writeFile: jest.fn(),
+  },
+}));
+
+jest.mock("prettier", () => ({
+  resolveConfig: jest.fn(),
+  format: jest.fn(),
+}));
+
+describe("formatWithPrettier", () => {
+  const mockReadFile = fs.readFile as jest.MockedFunction<typeof fs.readFile>;
+  const mockWriteFile = fs.writeFile as jest.MockedFunction<typeof fs.writeFile>;
+  const mockResolveConfig = resolveConfig as jest.MockedFunction<typeof resolveConfig>;
+  const mockFormat = format as jest.MockedFunction<typeof format>;
+
+  const mockResults: ESLint.LintResult[] = [
+    { filePath: "package.json" } as ESLint.LintResult,
+    { filePath: "hello-pepr.ts" } as ESLint.LintResult,
```

Comment:
What benefit do we gain from passing a collection of 2 elements as input?

Reply:
no

Reply:
Read it wrong like - "do we". I don't think that this detracts from the overall flow and unit tests are cheap. More of a nit but if you feel strongly..

Reply:
the function does receive an array of `ESLint.LintResults[]` too

Reply:
This input aligned with the assertions make me think the second assertion isn't really doing much aside from following the same code path as the entry before it:

```
    expect(mockWriteFile).toHaveBeenCalledWith("package.json", "const x = 1;");
    expect(mockWriteFile).toHaveBeenCalledWith("hello-pepr.ts", "const x = 1;");
```

Since both input files expect the same output, that signals it really doesn't matter what the arguments are since we're triggering the mock twice along the same execution path. Not a big enough deal to hold up the PR, though. I'll approve it and leave comment resolution (i.e., using one vs. two elements for test input) up to you.